src/Android/Xamarin.Android/Samples/Layers/KmlNetworkLinks/KmlNetworkLinks.cs

```diff
@@ -0,0 +1,171 @@
+                myKmlLayer.LoadStatusChanged += KmlLayer_LoadStatusChanged;
+
+                // Ensure the KML layer is loaded
+                await myKmlLayer.LoadAsync();
+
+                // Add the KML layer to the operational layers
+                _myMap.OperationalLayers.Add(myKmlLayer);
+
+                // Define and extent to zoom to (in the case: Frankfurt Germany area)
+                Envelope myExtent = new Envelope(647571.091263907, 6310478.90141415, 1350638.62905514, 6604034.01050928, SpatialReferences.WebMercator);
+
+                // Zoom to the extent
+                await _myMapView.SetViewpointGeometryAsync(myExtent);
+
+                // Traverse the file and display the refresh intervals for network links
+                TraverseNodesUpdateStatus(_myKmlDataset.RootNodes);
+            }
+            catch (Exception ex)
+            {
+                var builder = new AlertDialog.Builder(this);
+                builder.SetMessage("An error occurred" + ex.ToString()).SetTitle("Sample error").Show();
+            }
+        }
+
+        private void KmlLayer_LoadStatusChanged(object sender, Esri.ArcGISRuntime.LoadStatusEventArgs e)
+        {
+            if (e.Status == Esri.ArcGISRuntime.LoadStatus.Loaded)
+            {
+                // This makes sure that we switch to the UI thread before interacting with any UI components
+                RunOnUiThread(() =>
+                {
+                    // Clear the existing text view
+                    _textview_Label1.Text = "";
+
+                    // Traverse the file and display the refresh intervals for network links
+                    TraverseNodesUpdateStatus(_myKmlDataset.RootNodes);
+                });
+            }
+        }
+
+        private void TraverseNodesUpdateStatus(IReadOnlyList<KmlNode> sublayers)
+        {
+            // Iterate through each sublayer content
+            foreach (KmlNode content in sublayers)
+            {
+                // Update the UI if the content is a Network Link
+                if (typeof(KmlNetworkLink) == content.GetType())
+                {
+                    // Get a reference to the item as a KmlNetworkLink
+                    KmlNetworkLink netLink = content as KmlNetworkLink;
+
+                    // Get the existing text of the text view
+                    string existingTextBlock = _textview_Label1.Text as string;
+
+                    // Update the text with information about the layer
+                    existingTextBlock += String.Format("Name: {0} - Refresh Interval: {1}" +
+                        System.Environment.NewLine, netLink.Name, netLink.RefreshInterval);
+
+                    // Update the text view with the new text
+                    _textview_Label1.Text = existingTextBlock;
+                }
+
+                // Update the UI if the content is a Network Link
+                if (typeof(KmlNetworkLink) == content.GetType())
```

Comment:
This condition is the same as the condition above. The two blocks should probably be combined. 

Also, I believe there are other KmlNode types with children, including KmlDocument and KmlFolder. For a full exploration of the document, the recursion should happen for all KmlNodes with children.